DS-3404: JSPUI: Fix authority lookup

AlexanderS opened the pull request:
https://jira.duraspace.org/browse/DS-3404

This is #1578 against dspace-6_x slightly reworked and with an additional fix for `edit-item-form.jsp` like mentioned in the original pull request.

Commit:
JSPUI: Fix DSpaceChoiceLookup in edit-metadata.jsp

Commit:
JSPUI: Fix DSpaceChoiceLookup in edit-item-form.jsp

The collection id changed from a simple integer to a UUID and JavaScript
needs quotes around it.

tdonohue reviewed (approved):
👍 Code looks correct. Seems obvious that some single quotes are missing here. That said, I haven't tested it yet.

ihausmann committed:
DS-3404: JSPUI: Fix authority lookup

My colleague Eike made a mistake fixing the bug. His solution only works for repeatable fields.
 
https://jira.duraspace.org/browse/DS-3404
https://github.com/DSpace/DSpace/pull/1755

kshepherd reviewed (approved):
+1, yep, collection UUIDs will have introduced the need to quote the result of getID()
